code/messaging/src/phone/java/com/adobe/marketing/mobile/MessagingInternal.java

```diff
@@ -514,47 +520,69 @@ public void handleTrackingInfo(final Event event) {
+        proposition.put(ID, propositionInfo.id);
+        proposition.put(SCOPE, propositionInfo.scope);
+        proposition.put(SCOPE_DETAILS, propositionInfo.scopeDetails);
+        propositions.add(proposition);
+
+        final Map<String, Integer> propositionEventType = new HashMap<>();
+        propositionEventType.put(eventType.getPropositionEventType(), 1);
+
+        final Map<String, Object> decisioning = new HashMap<>();
+        decisioning.put(PROPOSITION_EVENT_TYPE, propositionEventType);
+        decisioning.put(PROPOSITIONS, propositions);
+
+        // add propositionAction if this is an interact eventType
+        if (eventType.equals(MessagingEdgeEventType.IN_APP_INTERACT)) {
+            final Map<String, String> propositionAction = new HashMap<>();
+            propositionAction.put(ID, interaction);
+            propositionAction.put(LABEL, interaction);
+            decisioning.put(PROPOSITION_ACTION, propositionAction);
+        }
 
-        // Create XDM data with tracking data
+        // create experience map with proposition tracking data
+        final Map<String, Object> experienceMap = new HashMap<>();
+        experienceMap.put(DECISIONING, decisioning);
+
+        // create XDM data with experience data
         final Map<String, Object> xdmMap = new HashMap<>();
-        final Map<String, Object> xdm = (Map<String, Object>) message.details.get(_XDM);
-        final Map<String, Object> xdmMixinsMap = (Map<String, Object>) xdm.get(MIXINS);
-        final Map<String, Object> experienceXdmMap = (Map<String, Object>) xdmMixinsMap.get(EXPERIENCE);
         xdmMap.put(XDMDataKeys.EVENT_TYPE, eventType.toString());
-        xdmMap.put(MessagingConstants.TrackingKeys.EXPERIENCE, experienceXdmMap);
-
-        // add iam mixin information if this is an interact eventType
-        if (eventType.equals(MessagingEdgeEventType.IN_APP_INTERACT) && !StringUtils.isNullOrEmpty(interaction)) {
-            final Map<String, Object> actionMap = new HashMap<>();
-            actionMap.put(XDMDataKeys.ACTION, interaction);
-            xdmMap.put(XDMDataKeys.IN_APP_NOTIFICATION_TRACKING_MIXIN_NAME, actionMap);
-        }
-
-        // Create the Meta Map
-        final Map<String, Object> metaMap = new HashMap<>();
-        final Map<String, Object> collectMap = new HashMap<>();
-        collectMap.put(DATASET_ID, datasetId);
-        metaMap.put(COLLECT, collectMap);
+        xdmMap.put(MessagingConstants.TrackingKeys.EXPERIENCE, experienceMap);
 
-        // Create xdm edge event data
-        final Map<String, Object> xdmData = new HashMap<>();
-        xdmData.put(XDM, xdmMap);
-        xdmData.put(META, metaMap);
+        // create maps for event history
+        final Map<String, String> iamHistoryMap = new HashMap<>();
+        iamHistoryMap.put(EVENT_TYPE, eventType.getPropositionEventType());
+        iamHistoryMap.put(MessagingConstants.EventMask.Keys.MESSAGE_ID, propositionInfo.activityId);
```

Comment:
why not import static `MessagingConstants.EventMask.Keys.MESSAGE_ID` like the rest of them?

Reply:
ah, i think i had a conflict previously where there was another static import also named MESSAGE_ID. that conflict no longer exists so i'm now importing the constant as well.